src/jit/regset.cpp

```diff
@@ -2666,17 +2666,24 @@ void RegSet::rsUnspillRegPair(GenTreePtr tree, regMaskTP needReg, KeepReg keepRe
 
         if (rsIsTreeInReg(regHi, tree))
         {
-            /* Temporarily lock the high part */
-
-            rsLockUsedReg(genRegMask(regHi));
+            // Temporarily lock the high part if necessary. If this register is a multi-use register that is shared
+            // with another tree, the register may already be locked.
+            const regMaskTP regHiMask = genRegMask(regHi);
```

Comment:
[This code](https://github.com/dotnet/coreclr/blob/master/src/jit/codegenlegacy.cpp#L3788-L3794) locks the registers used by a conditional jump's `op2` before unspilling its `op1`. These operands may share a register if that register is multi-use, so that register may be locked when we enter this function.